src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs

```diff
@@ -231,7 +316,7 @@ public static class ILoggerExtensions
 
             logger.LogWarning(DependencyFormat, new DependencyLogEntry(
                 dependencyType, 
-                dependencyName: null, 
+                dependencyName: targetName, 
```

Comment:
This is one I am not sure about, not sure if targetName is the right value for the dependencyName. 
Maybe it is better to add an overload where dependencyName can be specified and otherwise fall back on targetName?

Reply:
Agreed, then this one can call the new one and pass the targetName indeed but I'd consider deprecating this extension in favor of the new one.

Reply:
Added an overload so we can specify the dependencyName on the LogDependency, added some tests as well, see https://github.com/arcus-azure/arcus.observability/pull/229/commits/058865f2848428ff361e7ba2c033bae9becd83e1